src/Compilers/CSharp/Portable/Parser/CharacterInfo.cs

```diff
@@ -23,6 +23,16 @@ internal static bool IsHexDigit(char c)
         }
 
         /// <summary>
+        /// Returns true if the Unicode character is a binary (0-1) digit.
+        /// </summary>
+        /// <param name="c">The Unicode character.</param>
+        /// <returns>true if the character is a binary digit.</returns>
+        internal static bool IsBinaryDigit(char c)
+        {
+            return c == '0' || c == '1';
```

Comment:
This may be frequent, so I recommend avoiding control-flow. Try `|` instead of `||`.